compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1967,6 +1967,7 @@ object Types {
     def derivedSelect(prefix: Type)(implicit ctx: Context): Type =
       if (prefix eq this.prefix) this
       else if (prefix.isBottomType) prefix
+      else if (prefix.isInstanceOf[WildcardType]) WildcardType
```

Comment:
There's what look like an identical case at the end of this method (`case _: WildcardType => WildcardType`), can the two be merged?